### ISSUE TITLE
Ensure accessible target size for controls

### ISSUE THREAD

**Issue by danallsop** [ScottLogic/jumpstart-Leaflet#306]
> **Migrated from:** https://github.com/Leaflet/Leaflet/issues/7549
> **Original author:** danallsop
> **Original created:** 2021-04-10T12:23:19Z
---

## Motivation

The <span title="Web Content Accessibility Guidelines">WCAG</span> 2.1 <span title="Success Criterion">SC</span> [2.5.5 Target Size](https://www.w3.org/TR/WCAG21/#target-size) requires targets for [_pointer input_](https://www.w3.org/TR/WCAG21/#dfn-pointer-inputs) to be at least **44 by 44 in CSS pixels**. Meeting this requirement has [multiple benefits](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html#benefits), it'd be great to pass this criteria for Leaflet controls.

Edit: WCAG 2.2 (though it should be noted that the 2.2 version has not reached Recommendation status yet) has a more forgiving (from a designer's point of view!) Success Criterion: [2.5.8 Target Size (Minimum)](https://www.w3.org/TR/WCAG22/#target-size-minimum) which instead requires these targets to be at least **24 by 24** in CSS pixels.

## Proposed solution

Increase the size of the following controls to 44x44 px (or 24x24 per WCAG 2.2 SC 2.5.8).

- [ ] General controls:
  https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/dist/leaflet.css#L284-L286<br>
- [ ] Markers:
  https://github.com/Leaflet/Leaflet/blob/bc918d4bdc2ba189807bc207c77080fb41ecc196/src/layer/marker/Icon.Default.js#L26
  (Changing the marker's `width` and `height` to `44px` would cause it to stretch. Either the image would need to be modified or `padding` (right and left) could be modified to achieve a 44 by 44 px target.)<br><br>
- [ ] Popup's Close button:
  https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/dist/leaflet.css#L517-L518

## Additional context

Increasing the default size of controls would also have the benefit of not needing extra CSS for conditional `leaflet-touch` sizing (presumably, since the controls would be large enough by default). For example, the following lines could be removed:

https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/dist/leaflet.css#L316-L320

https://github.com/Leaflet/Leaflet/blob/0f904a515879fcd08f69b7f51799ee7f18f23fd8/dist/leaflet.css#L359-L362

**Comment by danallsop** [comment]
## Comments (1 total)

---
> **Comment by:** danallsop **on** 2021-04-10T12:45:02Z

In any case 44 by 44 px cannot be achieved for these targets (for example, for markers it can get tricky if increasing the size of the image itself is not desired, but setting a `padding` is also not desired as it may cause markers to be unexpectedly interacted with since they're larger than percieved) it should be added as a known limitation in the [accessibility statement](https://github.com/Leaflet/Leaflet/issues/7537).
